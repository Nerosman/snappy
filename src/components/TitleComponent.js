import React from "react";

export const TitleComponent = () => {
    switch (window.location.pathname) {
        case '/ship-info':
            return "Fill shipping info to receive a gift";
        case '/admin-panel':
            return "Snappy gifts order list";
        default:
            return "Snappy"
    }
};