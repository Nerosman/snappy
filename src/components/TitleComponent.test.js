import {configure, shallow} from "enzyme/build";
import React from "react";
import Adapter from "enzyme-adapter-react-16/build";
import {TitleComponent} from "./TitleComponent";

configure({ adapter: new Adapter() });

describe('<TitleComponent />', () => {
    test('take snapshot for navItemsComponent', () => {
        const wrapper = shallow(
            <TitleComponent />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should work with a switch statement with /ship-info pathname', () => {
        const wrapper = shallow(<TitleComponent pathname={"/ship-info"} />);
        expect(wrapper);
    });

    it('should work with a switch statement with /admin-panel pathname', () => {
        const wrapper = shallow(<TitleComponent pathname={"/admin-panel"} />);
        expect(wrapper);
    });
});
