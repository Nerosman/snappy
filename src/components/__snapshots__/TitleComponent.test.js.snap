// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TitleComponent /> take snapshot for navItemsComponent 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TitleComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): "Snappy",
  Symbol(enzyme.__nodes__): Array [
    "Snappy",
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
